### DISCUSSION TITLE
Convert io.debug into console.debug for JavaScript

### DISCUSSION THREAD

**Discussion by Arian94** [gleam-lang/gleam#3283] [Ideas & suggestions]
Isn't it a better idea to convert the ```io.debug``` to its counterpart in JS which is ```console.debug```?
ATM, it outputs ```console.log```.

**Comment by lpil** [comment]
It writes to stderr when possible. We could call console.debug when that isn't available. What would you see as being the advantages?

**Comment by Arian94** [comment]
Your focus is on NodeJs and Deno. If working on a browser, there will be
differences.

```console.debug``` only works if the Verbose mode is active in the
DevTools. I think it would be a small improvement on integrating Gleam and
JS if io.debug translates to console.debug.

**Reply by lpil** [reply]
Then not being printed by default seems like it would be a disadvantage, and it would mean that the function has a different behaviour in the browser compared to other runtimes where it would always print.

**Comment by Arian94** [comment]
It's not a disadvantage.

In NodeJs and Deno it will always log the meesage.
In a browser, a front-end developer may want to see some messages only in
the Verbose mode. That's why io.debug becomes useful. I don't want to see
my less-important messages be shown in the Default mode.

If a message is desired to be shown as usual the io.print would be the
proper function.

**Reply by lpil** [reply]
Having inconsistent behaviour is not an advantage.

**Reply by Arian94** [reply]
Good luck.
